solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -403,8 +402,9 @@ public void process(ResponseBuilder rb) throws IOException {
 
     req.getContext().put(SolrIndexSearcher.STATS_SOURCE, statsCache.get(req));
 
+    // TODO QueryResult ought not to be created and passed in methods of QueryComponent.
+    //  Should be created ~exclusively in SolrIndexSearcher and returned by it.
```

Comment:
I tend to not be a huge fan of comments in the code, what does this mean? it's  just for your convenience in the draft?
If that's the case, ignore my comment